### PR TITLE
Golang ci lint action

### LINKED ISSUE #648: [CI] - Migrate golangci-lint to official github action
Currently, we install and run golangci-lint by shell command:
https://github.com/OT-CONTAINER-KIT/redis-operator/blob/26f9c0be576c6fbcd263f322ac7bd38478aa8e5e/.github/workflows/operator-ci.yaml#L48-L61

We could change it to official action, see https://golangci-lint.run/usage/install/#github-actions

### PR DESCRIPTION
**Pull request by titanventura**
**Description**
Change `golangci-lint` from manual installation to gh-actions

Fixes #648 

**Type of change**
* New feature (non-breaking change which adds functionality)

**Checklist**
- [x] Functionality/bugs have been confirmed to be unchanged or fixed.
- [x] I have performed a self-review of my own code.

**Additional context**

Take a look at https://github.com/titanventura/ot-container-kit-redis-operator/actions/runs/6395766402 and https://github.com/titanventura/ot-container-kit-redis-operator/actions/runs/6395703984 to know more about how the change fits in !

### CODE REVIEW

File: .github/workflows/operator-ci.yaml [hunk 1]
```diff
@@ -66,11 +66,11 @@ jobs:
...
         with:
           go-version: ${{ env.GolangVersion }}
-      - name: Install GolangCI-Lint
-        run: |
-          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.40.0
       - name: Run GolangCI-Lint
-        run: ./bin/golangci-lint run --timeout 5m0s ./...
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.40.0
```
Comment (shubham-cmyk) [inline .github/workflows/operator-ci.yaml:72]:
```suggestion
          version: v1.54.0
```

File: .github/workflows/operator-ci.yaml [hunk 2]
```diff
@@ -66,11 +66,11 @@ jobs:
         uses: actions/setup-go@v2
```
Comment (shubham-cmyk) [inline .github/workflows/operator-ci.yaml:66]:
```suggestion
        uses: actions/setup-go@v4
```

Comment (shubham-cmyk) [inline .github/workflows/operator-ci.yaml:66]:
```suggestion
        uses: actions/setup-go@v4
```

**Comment by shubham-cmyk** [comment]
@titanventura Please make the required change.

**Comment by titanventura** [comment]
@shubham-cmyk , Have pushed the recommended changes, but golangci-lint reports some issues, can I solve those issues and bring it as a part of the same PR ?

**Comment by shubham-cmyk** [comment]
> @shubham-cmyk , Have pushed the recommended changes, but golangci-lint reports some issues, can I solve those issues and bring it as a part of the same PR ?

Okay

**Review by shubham-cmyk — APPROVED** [review]
Good work

**Pull request merged**